docs/guide/deployer.md

```diff
@@ -27,6 +27,22 @@ You can write deployment tasks synchronously and they'll be executed in the corr
 
 To deploy an ASA you must have `asa.yaml` file in `assets` folder. Head to the [ASA Definition File Spec](https://paper.dropbox.com/doc/Algorand-builder-specs-Vcdp0XNngizChyUWvFXfs#:uid=077585002872354521007982&h2=ASA-Definition-File) to learn more.
 
+#### Deploying ASA with custom parameters
+
+User can also override the fields in `assets/asa.yaml` when deploying ASA. Eg. If user wants to use a multisig address from 3 accounts: `alice, bob, john` as the Asset Reserve. We can create a multisig address in `algob` script first and then pass this address as a custom asaParams.
+
+```javascript
+const { createMsigAddress } = require('@algo-builder/algob');
+
+const addrs = [alice.addr, bob.addr, john.addr];
+const [mparams, multsigaddr] = createMsigAddress(1, 2, addrs); // version = 1, threshold = 2
+
+// while deploying ASA pass custom asa param
+await deployer.deployASA("ASA-2", {...}, { reserve: multsigaddr }); // this will overwrite reserve field in assets/asa.yaml
```

Comment:
how about merging it with ASADeployment flags ? this seems as extra variable here, adding them in ASADeployment flags seems clear to me

Reply:
I think passing it as a separate field is better. As `asaDef` has many fields so passing separately looks better to me. But if that doesn't better we can pass in deployment  flags only. let me know.